testcases/gcov_setup.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2017
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+'''
+gcov
+----
+
+A "Test Case" for extracting GCOV code coverage data from skiboot. The real
+use of this test case and code is to help construct code coverage reports
+for skiboot.
+'''
+
+import os
+import tempfile
+import unittest
+
+import OpTestConfiguration
+from common.OpTestSystem import OpSystemState
+from common.Exceptions import CommandFailed
+from common import OpTestInstallUtil
+from common.OpTestUtil import OpTestUtil
+
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class gcov(unittest.TestCase):
+    '''
+    Install the distro src  kernel and gcov is enabled and booted with
+    newly installed kernel
+    '''
+    def setUp(self):
+        self.conf = OpTestConfiguration.conf
+        self.cv_HOST = self.conf.host()
+        self.cv_SYSTEM = self.conf.system()
+        self.util = OpTestUtil(OpTestConfiguration.conf)
+        self.cv_HOST.host_run_command("echo 'set enable-bracketed-paste off' > .inputrc")
+        self.cv_HOST.host_run_command("bind 'set enable-bracketed-paste off'")
+        self.host_ip = self.conf.args.host_ip
+        self.host_user = self.conf.args.host_user
+        self.host_password = self.conf.args.host_password
+
+
+    def runTest(self):
+        '''
+        Running the gcov test
+        '''
+        log.info("OS: %s" %self.util.distro_name())
+        log.info("\nInstalling the ditro src...")
+        src_path = self.util.get_distro_src('kernel', '/root', "-bp")
+        out = self.cv_HOST.host_run_command(f"ls {src_path}")
+        for line in out:
+            if line.startswith("linux-"):
+                src_path = os.path.join(src_path, line)
+                break
+        log.info("\n\nsource path = %s" %src_path)
+        log.info("\nadding gcov_param....")
+        self.kernel_config(src_path)
+        log.info("Building the new kernel...")
+        self.build_and_boot(src_path)
+        log.info("rebooting the lpar after building the src with kernel parameters")
+        self.cv_HOST.host_run_command("reboot -f")
```

Comment:
this reboot will basically end the script in inconsistent state,  with exceptions... can we gracefully end the  test

Reply:
explore 
self.cv_SYSTEM.goto_state(OpSystemState.OFF)
self.cv_SYSTEM.goto_state(OpSystemState.OS)